tableau_blueprints/download_view.py

```diff
@@ -0,0 +1,157 @@
+import argparse
+import sys
+import os
+
+import tableauserverclient as TSC
+
+EXIT_CODE_FINAL_STATUS_SUCCESS = 0
+EXIT_CODE_UNKNOWN_ERROR = 3
+EXIT_CODE_INVALID_CREDENTIALS = 200
+EXIT_CODE_INVALID_RESOURCE = 201
+EXIT_CODE_FILE_WRITE_ERROR = 1021
+
+def get_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--username', dest='username', required=True)
+    parser.add_argument('--password', dest='password', required=True)
+    parser.add_argument('--site-id', dest='site_id', required=True)
+    parser.add_argument('--server-url', dest='server_url', required=True)
+    parser.add_argument('--view-name', dest='view_name', required=True)
+    parser.add_argument('--file-type', choices=['image', 'thumbnail', 'pdf', 'csv'], type=str.lower, required=True)
+    parser.add_argument('--file-name', dest='file_name', required=True)
+    parser.add_argument('--file-options', dest='file_options', required=False)
+    parser.add_argument('--workbook-name', dest='workbook_name', required=True)
+    args = parser.parse_args()
+    return args
+
+
+def authenticate_tableau(username, password, site_id, content_url):
+    """TSC library to sign in and sign out of Tableau Server and Tableau Online.
+
+    :param username:The name of the user.
+    :param password:The password of the user.
+    :param site_id: The site_id for required datasources. ex: ffc7f88a-85a7-48d5-ac03-09ef0a677280
+    :param content_url: This corresponds to the contentUrl attribute in the Tableau REST API.
+    :return: connection object
+    """
+    try:
+        tableau_auth = TSC.TableauAuth(username, password, site_id=site_id)
+        server = TSC.Server(content_url, use_server_version=True)
+        server.version = '3.15'
+        connection = server.auth.sign_in(tableau_auth)
+    except Exception as e:
+        print(f'Failed to connect to Tableau.')
+        print(e)
+        sys.exit(EXIT_CODE_INVALID_CREDENTIALS)
+    return server, connection
+
+
+def validate_get_view(server, connection, view_name, workbook_name):
+    """Validate workbook and returns the details of a specific view.
+
+    :param server:
+    :param connection: Tableau connection object
+    :param view_name: The name of the view.
+    :param workbook_name: The name of the workbook associated with the view.
+    :return: view_id: details of a specific view.
+    """
+    try:
+        view_id = None
+        req_option = TSC.RequestOptions()
+        req_option.filter.add(TSC.Filter(TSC.RequestOptions.Field.Name,
+                                         TSC.RequestOptions.Operator.Equals,
+                                         workbook_name))
+        with connection:
+            all_workbooks = server.workbooks.get(req_options=req_option)
+            workbook_id= [workbook.id for workbook in all_workbooks[0] if workbook.name == workbook_name]
```

Comment:
What happens if there are no workbooks returned and you access index 0 on line 66 from the all_workbooks variable? Are you relying on the general Exception catch for that - if so it's probably better to check the length of the response and return and exit in a different manner versus raising an Exception we can avoid.